### PR TITLE
3.5.0

### PR DESCRIPTION
**Pull request by MarDiehl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

h5py 3.5.0 has been released.
It needs the mpi plugin for pytest to disable certain tests that don't run for MPI builds. Unfortunately, these tests fail for MPI builds even in serial runs, i.e. runs not launched with `mpiexec`. Since I don't know how to launch `run_tests.py` with `mpiexec`, starting the test suite is now done explicitly in `meta.yaml`.

### CODE REVIEW

**Comment by MarDiehl** [comment]
@conda-forge-admin, please rerender

**Comment by MarDiehl** [comment]
@conda-forge-admin, please rerender

**Comment by MarDiehl** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (jakirkham) [inline recipe/meta.yaml]: We need `sys.exit` for these to actually raise the exit code correctly. That said, why can't we leave these in `run_test.py`?

**Review by tacaswell — DISMISSED** [review]

**Comment by jakirkham** [comment]
@tacaswell the tests are not raise correctly. Please see comment above

**Comment by tacaswell** [comment]
Sorry about that 🐑 .  I did indeed miss that issue and was in the process of dismissing my review.

**Comment by jakirkham** [comment]
All good 🙂

Got bitten by the `sys.exit` piece elsewhere recently. So it's fresh in my mind

**Comment by MarDiehl** [comment]
It seems that the only solution to get this working is to invoke pytest directly. This allows to pass the `--with-mpi` option which skips the tests that do not work for MPI (https://pytest-mpi.readthedocs.io/en/latest/usage.html). I don't know how to do that in the conda recipe. Especially, I found no way to invoke `run_tests.py` with `mpiexec`.

I also double check with the h5py developers whether it is correct that these tests fail if h5py/hdf5 is _build_ with MPI support. It would be nicer if raising this error could be a runtime decision: https://github.com/h5py/h5py/issues/1997

File: recipe/meta.yaml
Comment (MarDiehl) [reply recipe/meta.yaml]: that was my attempt to make pytest aware that this is an MPI build.

**Comment by jakirkham** [comment]
Ah ok. Thanks for the clarification!

**Comment by MarDiehl** [comment]
> I also double check with the h5py developers whether it is correct that these tests fail if h5py/hdf5 is _build_ with MPI support. It would be nicer if raising this error could be a runtime decision: [h5py/h5py#1997](https://github.com/h5py/h5py/issues/1997)

The error is raised by HDF5, not h5py. So nothing can be done here

**Comment by MarDiehl** [comment]
@jakirkham, @minrk, @ocefpaf, @scopatz, @tacaswell: Can anyone of you have a look at this and finish the work? I'm not very familiar with writing recipes.

**Comment by h-vetinari** [comment]
@MarDiehl, I picked up your changes into #107, since I was picking up #106 anyway after it had grown conflicts with master. l'd be happy to give you push access to my fork of this feedstock so we can collaborate on that PR.

**Comment by tacaswell** [comment]
Closing this as we have a 3.6 build build merged.  Given that we never got 3.5 to build I think we can let this go (I can not think of a reason why someone would _need_ a h5py35 build).

Thank you for your work on this @MarDiehl !

**Pull request closed without merging**
